storybookjs/storybook
# Vite: Fix stats-plugin to normalize file names with posix paths

Closes #27172

## What I did

Updates the `normalize` function within the `webpack-stats-plugin` for `builder-vite` to always normalize file names with posix path separators (`/`). 

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

1. On a window environment, create a storybook sandbox using react vite
2. Build storybook using `storybook build --stats-json`
3. Confirm the `name` and `id` fields in the `preview-stats.json` file are using posix path separators (`/`) and not windows separators (`\\`)

### Documentation

      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
